frontend/src/components/Collections/components/Dataset/components/DownloadDataset/components/Content/components/CurlLink/index.tsx

```diff
@@ -9,30 +9,18 @@ interface Props {
 }
 
 const CurlLink: FC<Props> = ({ fileName, handleAnalytics, link }) => {
-  const [isCopied, setIsCopied] = useState(false);
-
   const curl = `curl -o ${fileName} "${link}"`;
-
-  const handleCopyClick = () => {
-    setIsCopied(true);
-    copy(curl);
-    handleAnalytics();
-  };
-  const handleCopyMouseEnter = () => setIsCopied(false);
-
   return (
     <>
-      <CodeWrapper>
-        <Code>{curl}</Code>
-        <CodeMask onClick={handleCopyClick} onMouseEnter={handleCopyMouseEnter}>
-          {isCopied ? "Copied!" : "Copy to Clipboard"}
-        </CodeMask>
-      </CodeWrapper>
-      <Tip>
+      <CodeBlock>
+        <code>{curl}</code>
+        <CopyButton curl={curl} handleAnalytics={handleAnalytics} />
```

Comment:
Copy state and analytics functionality, with the addition of copying animation are relocated to a new component `CopyButton`.